### ISSUE TITLE
Parallel processing not working

### ISSUE THREAD

**Issue by sschooler** [glmmTMB/glmmTMB#910]
When running parallel processing using glmmTMB as described [here](https://cran.r-project.org/web/packages/glmmTMB/vignettes/parallel.html) through RStudio, R does not open up parallel clusters and instead serializes the model (task manager screenshot pasted below). 

```
N <- 4e6
xdata <- rnorm(N, 1, 2)
ydata <- 0.3 + 0.4*xdata + rnorm(N, 0, 0.25)
> system.time(model1 <- glmmTMB(formula = ydata ~ 1 + xdata, +  control = glmmTMBControl(parallel = 1)))    
user  system elapsed 
 20.42    1.05   22.78
> system.time(model2 <- glmmTMB(formula = ydata ~ 1 + xdata, + control = glmmTMBControl(parallel = 5)))   
  user  system elapsed 
  29.53   10.36   18.25 
```

I attempted to problem solve using [this thread](https://github.com/glmmTMB/glmmTMB/issues/620) to check connection with openMP. It seems like OMP is working, but I can't tell.
```
> glmmTMB::omp_check()
[1] TRUE
> TMB::openmp()
glmmTMB 
      1 
attr(,"autopar")
[1] FALSE
```

![Screenshot 2023-04-26 115407](https://user-images.githubusercontent.com/31750637/234676310-f50c26be-2f69-47fe-9e76-2492dccd6367.png)

Specs of my machine:
14 cores with 20 logical processors
16 GB RAM
Microsoft Windows 11 version 10.0.22621

R Studio Version 2023.03.0 Build 386
R version 2.3.0
glmmTMB version 1.1.7

**Comment by bbolker** [comment]
Hmm. I get

```r
   user  system elapsed 
 26.272   3.630  29.472 
```
in serial

and

```r
   user  system elapsed 
 61.138  10.133  16.919 
```
in parallel.

Are you **really** using R version 2.3.0? I doubt it.  4.3.0?

Can you try it in an R console rather than via RStudio, in case RStudio is part of the problem?

Here is an edited version of my `sessionInfo()`:

```
R Under development (unstable) (2023-04-21 r84297)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Pop!_OS 22.04 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

other attached packages:
[1] glmmTMB_1.1.7 peakRAM_1.0.2
```

(and TMB 1.9.4 is loaded via namespace).

**Comment by sschooler** [comment]
You are correct, it was R version 4.3. I think that I was/am running into a memory issue, and that GLMMtmb doesn't implement parallel processing the way I have previously seen- usually in task manager it shows up that there are multiple r sessions running, but that doesn't seem to be the case, potentially due to the OpenMP implementation.

Either way, it is working now, I am not sure why. I'm running into 
```
Error in MakeADGradObject(data, parameters, reportenv, random = random,  : 
  Caught exception 'std::bad_alloc' in function 'MakeADGradObject'
```

when I run my full model, but it includes a spatial autocorrelation term with 10,000 datapoints, so I think it's definitely a memory issue.
Thanks!

**Comment by bbolker** [comment]
Closing, feel free to request re-opening if you think there's more to do.
